drivers/mfd/mfd_axp192.c

```diff
@@ -50,12 +153,495 @@ static int mfd_axp192_init(const struct device *dev)
 	return 0;
 }
 
+int mfd_axp192_gpio_func_get(const struct device *dev, uint8_t gpio, enum axp192_gpio_func *func)
+{
+	const struct mfd_axp192_config *config = dev->config;
+	int ret;
+	uint8_t reg_fnc;
+
+	if (gpio >= AXP192_GPIO_NUM_PINS) {
+		LOG_ERR("Invalid gpio (%d)", gpio);
+		return -EINVAL;
+	}
+
+	if (k_is_in_isr()) {
+		return -EWOULDBLOCK;
+	}
```

Comment:
this is translating GPIO API requirements to the mfd layer... wrong

Reply:
I See. Thank you.